### PR TITLE
Use outbox messages to render messages to the chat screen

### PR DESCRIPTION
**Pull request by kunall17**
**Screenshot:** 

Last 3 messages are when server not running

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by kunall17** [comment]
Changed the look, as showing the elapsed time was of no use as the timestamp was showing that only! 

screen shot 2017-08-11 at 12 25 57 am

@borisyankov Updated, ready for review!

File: src/users/userSelectors.js [hunk 1]
```diff
@@ -29,6 +30,13 @@ export const getAccountDetailsUser = createSelector(
   },
 );
 
+export const getUserByEmail = (users: any[], userEmail: string) =>
```
Comment (kunall17) [inline src/users/userSelectors.js]: I have created this function in multiple PR's so lets have this, as it will be used in many places!

**Comment by kunall17** [comment]
@borisyankov done with the changes!

**Comment by borisyankov** [comment]
A PR I just merged did conflict with this. If you rebase again, I think it is ready to merge.
If @Sam1301 has something to say on this code, before I merge, she is welcome to do so :)

**Comment by kunall17** [comment]
@borisyankov Done :)

**Comment by Sam1301** [comment]
(looking at the code)

**Comment by borisyankov** [comment]
... and it's merged.

File: src/message/OutboxMessageContainer.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+}
+
+export default connect(
+  state => ({
+    auth: getAuth(state),
+    twentyFourHourTime: state.realm.twentyFourHourTime,
+  }),
+  boundActions,
+)(connectActionSheet(OutboxMessageContainer));
```
Comment (Sam1301) [inline src/message/OutboxMessageContainer.js]: are we using `connectActionSheet`?

File: src/message/OutboxMessageContainer.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    twentyFourHourTime: boolean,
+    isBrief: boolean,
+  };
+
+  static defaultProps = {
+    twentyFourHourTime: false,
+  };
+
+  render() {
+    const { message, auth, actions, twentyFourHourTime, isBrief } = this.props;
```
Comment (Sam1301) [inline src/message/OutboxMessageContainer.js]: this component seems very similar to MessageContainer; maybe we make changes in MessageContainer itself to keep the code DRY?

File: src/message/OutboxMessageContainer.js (hunk 2 above)
Comment (kunall17) [inline src/message/OutboxMessageContainer.js]: I left it as, I thought I'll use this as menu for having menu's like retry or cancel message send, which should be a part of an outbox!

File: src/message/OutboxMessageContainer.js (hunk 3 above)
Comment (kunall17) [inline src/message/OutboxMessageContainer.js]: Initially I did that but it caused a lot of if condition's in the component, hence I decided to make it's own component!

File: src/message/MessageFull.js [hunk 4]
```diff
@@ -48,7 +49,15 @@ class MessageFull extends PureComponent {
...
   render() {
-    const { message, children, twentyFourHourTime, ownEmail, starred, onLongPress } = this.props;
+    const {
+      message,
+      children,
+      twentyFourHourTime,
+      ownEmail,
+      starred,
+      onLongPress,
+      isNotYetSent,
```
Comment (kunall17) [inline src/message/MessageFull.js]: isNotYetSent is used in the MessageTags

File: src/message/OutboxMessageContainer.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    const MessageComponent = isBrief ? MessageBrief : MessageFull;
+    const childrenNodes = htmlToDomTree(message.parsedContent);
+    return (
+      <MessageComponent
+        message={message}
+        twentyFourHourTime={twentyFourHourTime}
+        ownEmail={message.email}
+        outbox>
```
Comment (kunall17) [inline src/message/OutboxMessageContainer.js]: This should be named as `isNotYetSent`.
I should stop using find and replace for review changes :sweat_smile:

I'll address this one in a PR for the action sheet's!

**Comment by kunall17** [comment]
Thanks for the review btw @Sam1301 👍

**Comment by Sam1301** [comment]
Cool, I haven't done a thorough review, will do so latest by tomorrow 👍

**Pull request merged**